src/Bicep.Core/TypeSystem/Providers/ProviderSpecificationSyntaxFactory.cs

```diff
@@ -0,0 +1,82 @@
+
+public static partial class ProviderSpecificationSyntaxFactory
+{
+    // The setting below adds syntax highlighting for regex.
+    // language=regex
+    private const string NamePattern = "[a-zA-Z][a-zA-Z0-9]+";
+
+    private const string BicepRegistryAddressPattern = @"br[:\/]\S+";
+
+    // Regex copied from https://semver.org/.
+    // language=regex
+    private const string SemanticVersionPattern = @"(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?";
+
+    [GeneratedRegex(@$"^(?<name>{NamePattern})$", RegexOptions.ECMAScript | RegexOptions.Compiled)]
+    private static partial Regex ConfigManagedSpecificationPattern();
+
+    [GeneratedRegex(@$"^(?<name>{NamePattern})@(?<version>{SemanticVersionPattern})$", RegexOptions.ECMAScript | RegexOptions.Compiled)]
+    private static partial Regex LegacyBuiltInSpecificationPattern();
+
+    [GeneratedRegex(@$"^(?<address>{BicepRegistryAddressPattern})@(?<version>{SemanticVersionPattern})$", RegexOptions.ECMAScript | RegexOptions.Compiled)]
+    private static partial Regex InlinedSpecificationPattern();
+
+    [GeneratedRegex(@"^\S*[:\/](?<name>\S+)$", RegexOptions.ECMAScript | RegexOptions.Compiled)]
+    private static partial Regex RepositoryNamePattern();
+
+    public static IProviderSpecificationSyntax CreateProviderSpecificationSyntax(SyntaxBase syntax)
+     => syntax switch
+     {
+         StringSyntax stringSyntax when stringSyntax.TryGetLiteralValue() is { } value &&
+                                       TryCreateFromStringSyntax(stringSyntax, value) is { } specification
+             => specification,
+
+         IdentifierSyntax identifierSyntax
+             => new ConfigurationManagedProviderSpecificationSyntax(
+                 identifierSyntax.IdentifierName,
+                 IsValid: true,
+                 identifierSyntax.Span),
+
+         _ => new ProviderSpecificationSyntaxTrivia(syntax.Span)
+     };
+
+    private static IProviderSpecificationSyntax? TryCreateFromStringSyntax(StringSyntax stringSyntax, string value)
+    {
+        if (LegacyBuiltInSpecificationPattern().Match(value) is { Success: true } legacyMatch)
+        {
+            var name = legacyMatch.Groups["name"].Value;
+            var version = legacyMatch.Groups["version"].Value;
+            var span = new TextSpan(stringSyntax.Span.Position + 1, name.Length);
+
+            return new LegacyProviderSpecificationSyntax(name, version, IsValid: true, span);
+        }
+        else if (ConfigManagedSpecificationPattern().Match(value) is { Success: true } builtInMatch)
+        {
+            var name = builtInMatch.Groups["name"].Value;
+            var span = new TextSpan(stringSyntax.Span.Position + 1, name.Length);
+
+            return new ConfigurationManagedProviderSpecificationSyntax(name, IsValid: true, span);
+        }
```

Comment:
Is this form of the syntax something we plan to support?

Reply:
Nope, I will remove this clause :)